### DISCUSSION TITLE
Configure the Kafka custom resource to use a custom authorizer

### DISCUSSION THREAD

**Discussion by priemjs** [strimzi/strimzi-kafka-operator#10831] [Q&A] [answered]
[authorizer.txt](https://github.com/user-attachments/files/17740170/authorizer.txt)
I am looking for examples on how to write a custom authorizer within the framework of Strimzi.  I found documentation on how to configure Strimzi for a [custom authorizer](documentation/api/io.strimzi.api.kafka.model.kafka.KafkaAuthorizationCustom.adoc)  however can not find details on how to fulfill the kafka Authorizer interface.  Looking to expand on the[ kafka-oauth-strimz](https://github.com/strimzi/strimzi-kafka-oauth) project.  Any guidance would be beneficial.  

Attached is an attempt to implement an authorizer that allows always.  However the brokers will not connect when running under docker containers.

**Comment by scholzj** [comment, marked as answer]
The authorizer is a Kafka thing, not a Strimzi thing. Strimzi only configures it. So while you can of course use the Strimzi OAuth Authorizer as an example, the documentation of how to write a custom authorizer should be part of Apache Kafka.
